src/main/java/org/opentripplanner/standalone/config/sandbox/FlexConfig.java

```diff
@@ -1,31 +1,94 @@
 package org.opentripplanner.standalone.config.sandbox;
 
 import static org.opentripplanner.standalone.config.framework.json.OtpVersion.V2_1;
+import static org.opentripplanner.standalone.config.framework.json.OtpVersion.V2_3;
 
-import org.opentripplanner.ext.flex.FlexParameters;
-import org.opentripplanner.routing.api.request.preference.RoutingPreferences;
+import java.time.Duration;
 import org.opentripplanner.standalone.config.framework.json.NodeAdapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class FlexConfig {
 
-  public static final int DEFAULT_MAX_TRANSFER_SECONDS = 60 * 5; // 5 minutes
-  public final int maxTransferSeconds;
+  public static final FlexConfig DEFAULT = new FlexConfig();
 
-  public FlexConfig(NodeAdapter json) {
-    maxTransferSeconds =
+  private static final Duration MAX_TRANSFER_DURATION = Duration.ofMinutes(5);
+  private static final Duration MAX_FLEX_TRIP_DURATION = Duration.ofMinutes(45);
+
+  private static final Logger LOG = LoggerFactory.getLogger(FlexConfig.class);
+
+  private final Duration maxTransferDuration;
+  private final Duration maxFlexTripDuration;
+
+  private FlexConfig() {
+    maxTransferDuration = MAX_TRANSFER_DURATION;
+    maxFlexTripDuration = MAX_FLEX_TRIP_DURATION;
+  }
+
+  public FlexConfig(NodeAdapter root, String parameterName) {
+    var json = root
+      .of(parameterName)
+      .since(V2_1)
+      .summary("Configuration for flex routing.")
+      .asObject();
+
+    Duration maxTransferDuration;
+
+    if (json.exist("maxTransferDurationSeconds")) {
```

Comment:
I'm usually with Thomas here. People are just as likely to overlook the config change when you wait so you might as well make it backwards-incompatible now.

Reply:
I will remove the compatibility